src/middlewared/middlewared/plugins/vm.py

```diff
@@ -921,13 +947,17 @@ def random_mac(self):
         if verrors:
             raise verrors
 
-        devices = data.pop('devices', None)
-        if devices:
-            update_devices = await self.__do_update_devices(id, devices)
-        if data:
-            return await self.middleware.call('datastore.update', 'vm.vm', id, data)
-        else:
-            return update_devices
+        devices = new.pop('devices', [])
+        new.pop('status', None)
+        if devices != old['devices']:
+            success = await self.safe_devices_updates(devices)
+            if not success:
+                raise CallError('Could not create new resources, please check logs.')
```

Comment:
As before, would it be helpful to add a path/pointer to the log file location?

Reply:
Same here, why not raise the error?